sound/soc/sof/intel/pci-icl.c

```diff
@@ -27,6 +27,7 @@ static const struct sof_dev_desc icl_desc = {
 	.resindex_pcicfg_base   = -1,
 	.resindex_imr_base      = -1,
 	.irqindex_host_ipc      = -1,
+	.sdw_bidir_base		= 2,
```

Comment:
This is not really platform related, does it? It is IPC3 related, IPC4's bidir base is 0, right?

Reply:
> This is not really platform related, does it? It is IPC3 related, IPC4's bidir base is 0, right?

That is platform related. SDW bidirectional pins are the same on IPC3 and IPC4. But, IPC4 FW supports all SDW pins. So, we don't need to shift the dai index in IPC4.

Reply:
Right, so, if you boot cAVS2.5 (TGL/ADL/etc) the sdw_bidir_base is applicable, but it is applicable only if you boot to use IPC3. Same platform, different IPC version.
sdw_bidir_base is not platform related option, it is IPC version specific, no?

Reply:
or you are saying that with both IPC3/4 the topology's sdw bidir index starts at 2 and we don't need to normalize this in case of IPC4?

Or to put this to other way: On MTL+ the bidir_base is 0 in (IPC4) topology and it is 2 in (IPC4) cAVS2.5?

Reply:
sdw_bidir_base will be applied to IPC3 only because dai_index 4, and 5 are acceptable by IPC4. As far as I know we always use 2+ SDW Pins even on MTL+ platforms.

Reply:
you are not touching pci-mtl, pci-lnl, for those platforms the sdw_bidir_base is 0.

Why do you even need this in the `struct sof_dev_desc` when the subtraction is only applicable for INTEL_ALH with IPC3 and it is always applicable in this case?

If you are saying that this is somehow a platform variable and applicable to all Intel platforms, then you should set the correct value for all platforms that this applies.

Reply:
> you are not touching pci-mtl, pci-lnl, for those platforms the sdw_bidir_base is 0.
> 
> Why do you even need this in the `struct sof_dev_desc` when the subtraction is only applicable for INTEL_ALH with IPC3 and it is always applicable in this case?

Because AMD or other vendors may not need the offset.

> 
> If you are saying that this is somehow a platform variable and applicable to all Intel platforms, then you should set the correct value for all platforms that this applies.

Sure I will add sdw_bidir_base to all Intel platforms. We define SOC_SDW_INTEL_BIDIR_PDI_BASE 2 and apply it to all Intel platforms. But, that doesn't match to the HW capabilities. I am not sure about older platforms, but I checked TGL, MTL, and LNL, all SDW pins are bidirectional. You can check the HW capabilities by the dmesg log below.
```
soundwire_intel:intel_pdi_init: soundwire_intel soundwire_intel.link.0: PCM cap bd:12 in:0 out:0
```

Reply:
Added sdw_bidir_base to all Intel platforms.

Reply:
> AMD and other vendors don't use dai type `SOF_DAI_INTEL_ALH`?

Oh, yes, that is a valid point. Then I think we don't need sdw_bidir_base.

Reply:
updated